examples/applications/cloud-app/src/Program.cs

```diff
@@ -2,109 +2,482 @@
+                    }
+                    default:
+                    {
+                        Console.Error.WriteLine("Sorry, but this has not yet been implemented.");
+                        break;
+                    }
+                }
                 break;
             }
-            case "quit":
+            case IInputErrorResult:
             {
-                quit = true;
+                Console.Error.WriteLine("Invalid usage. Try one of the following:");
+                PromptArguments.PrintUsage(Console.Error);
                 break;
             }
-            case var line when Regex.Match(line, @"^subscribe\s+([^\s]+)$") is { Success: true, Groups: [ _, { Value: var st } ] }:
+        }
+
+        if (request is { } someRequest)
+        {
+            try
             {
-                var options =
-                    mqttFactory.CreateSubscribeOptionsBuilder()
-                               .WithTopicFilter(f => f.WithTopic(st))
-                               .Build();
-                await mqttClient.SubscribeAsync(options, CancellationToken.None);
-                break;
+                var response = await timeout.ApplyAsync(cancellationToken =>
+                    rpcClient.ExecuteAsync(session.Vin, someRequest, cancellationToken));
+
+                Console.WriteLine(response.ToJsonEncoding(jsonSerializerOptions));
             }
-            case var line when Regex.Match(line, @"^publish\s+([^\s]+)\s+(.+)$") is { Success: true, Groups: [ _, { Value: var pt }, { Value: var payload } ] }:
+            catch (OperationCanceledException ex)
             {
-                var message =
-                    new MqttApplicationMessageBuilder()
-                        .WithTopic(pt)
-                        .WithPayload(payload)
-                        .Build();
-                await mqttClient.PublishAsync(message, CancellationToken.None);
-                break;
+                Console.Error.WriteLine(ex.Message);
             }
-            default:
+            catch (ChariottRpcException ex)
             {
-                Console.Error.WriteLine($"Invalid command! Try one of the following:{Environment.NewLine}{help}");
-                break;
+                Console.Error.WriteLine(ex.Detail ?? ex.Message);
             }
         }
     }
 
-    await mqttClient.DisconnectAsync(mqttFactory.CreateClientDisconnectOptionsBuilder().Build(), CancellationToken.None);
+    await timeout.ApplyAsync(cancellationToken =>
+        mqttClient.DisconnectAsync(mqttFactory.CreateClientDisconnectOptionsBuilder().Build(), cancellationToken));
 
     return 0;
 }
-catch (Exception ex)
+
+static bool TryParseValue(string input,
+                          [NotNullWhen(true)] out Value.ValueOneofCase? kind,
+                          [NotNullWhen(true)] out Value? value)
 {
-    Console.Error.WriteLine(ex);
-    return 1;
+    input = input.Trim();
+    kind = default;
+    value = null;
+
+    if (input is "true")
```

Comment:
[nit] Should we be case-insensitive here?

Reply:
Everything else is case-sensitive (like all commands and options) so it would be odd to make this an exception.